main/main.cpp

```diff
@@ -3201,89 +3201,89 @@ int Main::start() {
 	main_timer_sync.init(OS::get_singleton()->get_ticks_usec());
 	List<String> args = OS::get_singleton()->get_cmdline_args();
 
-	for (int i = 0; i < args.size(); i++) {
+	for (List<String>::Element *E = args.front(); E; E = E->next()) {
 		// First check parameters that do not have an argument to the right.
 
 		// Doctest Unit Testing Handler
 		// Designed to override and pass arguments to the unit test handler.
-		if (args[i] == "--check-only") {
+		if (E->get() == "--check-only") {
```

Comment:
Similar to the earlier case I mentioned, might be better to do the `E->get()` once. The original was getting a little silly... :grin:

Reply:
Would need to make sure it covers changing cases with the iteration but good point, will add